src/main/java/use_case/summary/SummaryInteractor.java

```diff
@@ -13,16 +13,25 @@ public class SummaryInteractor implements SummaryInputBoundary {
     @Override
     public void execute(SummaryInputData inputData, SummaryAPIInterface APIWrapper) {
         Episode episode = inputData.getEpisode();
-        try {
-            String summary = APIWrapper.generateSummary(episode.getTranscript());
-            episode.setSummary(summary);
-            // once the summary is generated, set the summary and save the episode
-            episodeDataAccessObject.saveEpisode(episode);
-            SummaryOutputData outputData = new SummaryOutputData(episode, false);
-            outputBoundary.prepareSuccessView(outputData);
-        } catch (IOException e) {
+        if (episode.getSummary() != null) {
```

Comment:
There might be a small logical error in this if statement, maybe you need episode.getsummary == null?

Reply:
You're right, I made this change and forgot to commit it haha

Reply:
haha no worries